### ISSUE TITLE
Can't open object can't open my reads

### ISSUE THREAD

**Issue by FengqingWang0712** [GATB/simka#18]
Hi there,
I built an input file of my reads. It said can't read my reads with the same error when I ran simka. The example tested successfully. Could someone show me where the problem is? Thanks very much!

Here is the error: 

Creating input

 Nb input datasets: 1

HDF5-DIAG: Error detected in HDF5 (1.10.5) thread 0:

   #000: 

/scratch/fwang/simka-v1.5.3-Source/thirdparty/gatb-core/gatb-core/thirdparty/hdf5/src/H5A.c 

line 425 in H5Aopen(): unable to load attribute info from object header 

for attribute: 'version'

     major: Attribute

     minor: Can't open object

   #001: 

/scratch/fwang/simka-v1.5.3-Source/thirdparty/gatb-core/gatb-core/thirdparty/hdf5/src/H5Aint.c 

line 433 in H5A__open(): unable to load attribute info from object 

header for attribute: 'version'

     major: Attribute

     minor: Can't open object

   #002: 

/scratch/fwang/simka-v1.5.3-Source/thirdparty/gatb-core/gatb-core/thirdparty/hdf5/src/H5Oattribute.c 

line 515 in H5O__attr_open_by_name(): can't locate attribute: 'version'

     major: Attribute

     minor: Object not found

HDF5-DIAG: Error detected in HDF5 (1.10.5) thread 0:

   #000: 

/scratch/fwang/simka-v1.5.3-Source/thirdparty/gatb-core/gatb-core/thirdparty/hdf5/src/H5A.c 

line 704 in H5Aget_space(): not an attribute

     major: Invalid arguments to routine

     minor: Inappropriate type

HDF5-DIAG: Error detected in HDF5 (1.10.5) thread 0:

   #000: 

/scratch/fwang/simka-v1.5.3-Source/thirdparty/gatb-core/gatb-core/thirdparty/hdf5/src/H5S.c 

line 1013 in H5Sget_simple_extent_dims(): not a dataspace

     major: Invalid arguments to routine

     minor: Inappropriate type

HDF5-DIAG: Error detected in HDF5 (1.10.5) thread 0:

   #000: 

/scratch/fwang/simka-v1.5.3-Source/thirdparty/gatb-core/gatb-core/thirdparty/hdf5/src/H5A.c 

line 662 in H5Aread(): not an attribute

     major: Invalid arguments to routine

     minor: Inappropriate type

ERROR: Can't open dataset: ID1

Here are the IDs of my reads as input:

ID1: 2018031910_paired_1.fasta ; 2018031910_paired_2.fasta

ID2: 201803193_paired_1.fasta ; 201803193_paired_2.fasta

ID3: 20180319_paired_1.fasta ; 20180319_paired_2.fasta

ID4: 20180403_paired_1.fasta ; 20180403_paired_2.fasta

ID5: 20180405_paired_1.fasta ; 20180405_paired_2.fasta

ID6: 20180410_paired_1.fasta ; 20180410_paired_2.fasta

ID7: 2018041210_paired_1.fasta ; 2018041210_paired_2.fasta

ID8: 201804123_paired_1.fasta ; 201804123_paired_2.fasta

ID9: 20180412_paired_1.fasta ; 20180412_paired_2.fasta

ID10: 2018041710_paired_1.fasta ; 2018041710_paired_2.fasta

ID11: 201804173_paired_1.fasta ; 201804173_paired_2.fasta

ID12: 20180417_paired_1.fasta ; 20180417_paired_2.fasta

ID13: 20180419_paired_1.fasta ; 20180419_paired_2.fasta

ID14: 20180424_paired_1.fasta ; 20180424_paired_2.fasta

ID15: 2018042610_paired_1.fasta ; 2018042610_paired_2.fasta

ID16: 201804263_paired_1.fasta ; 201804263_paired_2.fasta

ID17: 20180426_paired_1.fasta ; 20180426_paired_2.fasta

ID18: 20180502_paired_1.fasta ; 20180502_paired_2.fasta

ID19: 20180503_paired_1.fasta ; 20180503_paired_2.fasta

ID20: 2018050810_paired_1.fasta ; 2018050810_paired_2.fasta

ID21: 201805083_paired_1.fasta ; 201805083_paired_2.fasta

ID22: 20180508_paired_1.fasta ; 20180508_paired_2.fasta

ID23: 2018051110_paired_1.fasta ; 2018051110_paired_2.fasta

ID24: 201805113_paired_1.fasta ; 201805113_paired_2.fasta

ID25: 20180511_paired_1.fasta ; 20180511_paired_2.fasta

ID26: 20180515_paired_1.fasta ; 20180515_paired_2.fasta

ID27: 20180517_paired_1.fasta ; 20180517_paired_2.fasta

ID28: 2018052210_paired_1.fasta ; 2018052210_paired_2.fasta

ID29: 201805223_paired_1.fasta ; 201805223_paired_2.fasta

ID30: 20180522_paired_1.fasta ; 20180522_paired_2.fasta

ID31: 20180524_paired_1.fasta ; 20180524_paired_2.fasta

ID32: 2018052910_paired_1.fasta ; 2018052910_paired_2.fasta

ID33: 201805293_paired_1.fasta ; 201805293_paired_2.fasta

ID34: 20180529_paired_1.fasta ; 20180529_paired_2.fasta

**Comment by clemaitre** [comment]
Hi,

This error occurs when the program does not find the files at the indicated path. 
Here, your input file contains only file names (and not absolute paths for instance), in this case, your fasta files must be in the same directory as the input file. Is it the case ?

**Comment by FengqingWang0712** [comment]
> Hi,
> 
> This error occurs when the program does not find the files at the indicated path.
> Here, your input file contains only file names (and not absolute paths for instance), in this case, your fasta files must be in the same directory as the input file. Is it the case?

Thanks for your reply. It seems not this case. The fasta files and the input files are in the same directory. I ran again after adding absolute paths in input files just now. It showed the same error. Are there specific requirements for the contigs headers? My contigs' headers are like this: >000000000011. Is there a possibility that simka didn't recognize my fasta files?

**Comment by clemaitre** [comment]
> Are there specific requirements for the contigs headers? My contigs' headers are like this: >000000000011. Is there a possibility that simka didn't recognize my fasta files?
No, headers can be anything you want. The error is typical of a "file not found". Are you 100% sure that your files are correctly spelled in the input file and also readable by your system or user ?

**Comment by FengqingWang0712** [comment]
> > Are there specific requirements for the contigs headers? My contigs' headers are like this: >000000000011. Is there a possibility that simka didn't recognize my fasta files?
> 
> No, headers can be anything you want. The error is typical of a "file not found". Are you 100% sure that your files are correctly spelled in the input file and also readable by your system or user?

Hi. Thanks for your reply. 
I checked my data and ran many times using different input.txt files. Luckily, I found the issue finally. It's because of my text format. I use another way to generate the input.txt with the same information and it's working now. I tried using the example txt file and it's proved that my text format is really the reason. It seems that my first text format can't be read by the program. Thanks very much! All the best!

**Comment by samuelanyaso** [comment]
I encountered the same issue. The program seems to have trouble recognizing plain .txt files. A quick way around this is to convert plain .txt files to UNIX format.
`dos2unix input_file.txt`

**Comment by FengqingWang0712** [comment]
> I encountered the same issue. The program seems to have trouble recognizing plain .txt files. A quick way around this is to convert plain .txt files to UNIX format.
> `dos2unix input_file.txt`

Thanks very much!
